hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MulticastJoiner.java

```diff
@@ -148,7 +141,7 @@ public void onMessage(Object msg) {
                 logger.warning(e);
             }
         } finally {
-            node.multicastService.removeMulticastListener(listener);
+            splitBrainJoinMessages.clear();
```

Comment:
I'm just thinking what if there's more brains than two, for example five? 
With this logic their join requests will be discarded. 
That is the reason I was thinking that we maybe need some message timestamp and expiration duration?